### PR TITLE
feat: Planetary outfits storage

### PR DESCRIPTION
**Pull request by petervdmeer**
This PR implements planetary storage for outfits;
- Outfits can be stored on a planet by selecting an outfit and pressing (u)ninstall or (r)emove in the outfitter screen.
- Re-installing outfits can be done using buy (first from cargo, then from storage, then from outfitter).
- Selling outfits can be done using normal sell (first from cargo, then from selected ships, then from storage).

Hereby a screenshot to show how it looks in the game; some items are in storage and some items are in stock and in cargo as well.
storage_example

Here is a screenshot from the map-popup to see where outfits are stored:
stored_outfits_popup

Here is a screeshot from the map with markers to show where outfits are stored:
stored_outfits_rays

Here is the same storage, but now taken from the savegame (with also some missiles stored on a different planet):
```
storage
	planet Mainsail
		cargo
			outfits
				"Systems Core (Large)" 1
				"Large Heat Shunt" 1
				"Outfits Expansion" 1
				"Wanderer Ramscoop" 1
				"Gatling Gun Ammo" 123
	planet Ada
		cargo
			outfits
				"Meteor Missile" 5
```

Feature request #724 has a nice description of why we would want this. Plugins and player-advice about storing outfits in large warehouse ships also suggest that this feature is desired. This feature was also discussed on Discord in the content-creating channel.

Not planned by me to be included in this PR, but nice additions for the future are:
- Get an overview list in the player-info screen section of where (which) outfits are stored.

### CODE REVIEW

**Comment by Zitchas** [comment]
Looks nice! Is there any kind of indicator on the map which planet one has stuff stored on?

**Comment by petervdmeer** [comment]
> Looks nice! Is there any kind of indicator on the map which planet one has stuff stored on?

No, not in this PR, but I agree that that would be a very desirable feature to have. When this PR gets merged, then that would probably be one of the next PRs I would pick-up.

**Comment by petervdmeer** [comment]
> Looks nice! Is there any kind of indicator on the map which planet one has stuff stored on?

Now there is (on the map, similar to parked escorts):
stored_outfits_popup

**Comment by Zitchas** [comment]
Excellent!

**Comment by beccabunny** [comment]
For a visual indicator on the map, I was thinking about this: 8 rays coming out from the system's circle, that finish in the "selected system" circle. So, in the mock up, polaris would have stored outfits+parked escorts and alpha hydri stored outfits+unparked escorts, and Kor Zena has stored outfits but not escorts.
Senzanome

**Comment by petervdmeer** [comment]
I updated the drawing. The stored outfits are now drawn like this:
stored_outfits_rays

**Comment by waterhouse** [comment]
First, this is a very nice feature.  It feels ridiculous to have a fleet of ships acting just as "beasts of burden", especially when I want them to go places that need lots of fuel or jump drives; then I'm incentivized to put the jump drives on ships with maximum outfit space, so I get a fleet of badly-armed Leviathans or Shield Beetles just shuttling stuff around.  And even with that, big engines are difficult to move around due to limited engine space; e.g. if a Leviathan has a Bufaer thruster, then I have 16 tons to give it steering.  This is much more reasonable, and it makes solo runs much more practicable.

With that said, here are issues I have:

1. It's not very discoverable.  Even though I applied the patch myself, I had to go back to the pull request to learn the "t" button.  Not sure the best way to present it—some kind of "?" mouseover thing that shows all the outfitter hotkeys?

2. Currently, if there are some items in stock, but none in the current ship or in cargo, then pressing "t" does nothing, in either normal or cargo mode.  (As a result, if you want to move items from stock into storage, you have to either put *all* of them onto your ship—all because "b" selects from storage before stock—or switch to cargo mode and buy and store items one by one.)  It should, instead, move items from stock to storage.

3. Not sure if it's as a result of these changes, but here is what I'd call a bug: I selected cargo mode in the outfitter.  There was an item in stock (not storage), which couldn't fit in my cargo.  I pressed "b".  The item disappeared from stock and didn't show up in cargo either.  It was gone forever.  This is easily reproducible.

I tried repeating the above, but with the items in storage rather than in stock—and in this case the item (heavy laser) was also in supply.  When I tried to buy in cargo mode, the game complained that I didn't have space for the item; the number of items in storage did not change, but my money was still spent, and also the sell price changed from 25% (due to depreciation) to 100%.  It's as if it allowed me to buy an item at full price, but then deleted one of my existing items forever.

4. If the "cargo view" is selected, there doesn't seem to be a button to move from storage to cargo (when there's none in stock)--neither "b" nor "i" works.  You have to sell so it's "in stock", then buy into cargo.  I would recommend at least making "i" work for that.

(As a general comment, with five different places outfits can be (supply, stock, installed, cargo, and now storage), and only a few buttons ("s", "b", "t", "i"), it takes care to ensure that all use cases can be handled simply.  But I think that the changes from (2) and (4) should suffice.)

**Comment by petervdmeer** [comment]
@waterhouse Thanks for the review. I'll look into it.

**Comment by waterhouse** [comment]
More on (3) above:
1. In the base game, in cargo mode, if you have no room to put an outfit, then the "buy" button is simply greyed out.
2. With your patch: if you're in cargo mode, and you have no room to put an outfit:
2a. If you have no items in storage, then the buy button is greyed out.
2b. If you have any other item (not necessarily the same outfit type) in storage, then the buy button is active (if you have enough cash) despite there being no room for the outfit.  If you click it (or press "b"), then your cash is spent but the item (if it had a finite number in stock) disappears.

I've attached a pilot file to illustrate it—created from a version without your changes, based on current git.  It's a starter shuttle with four energy blasters in cargo bay (hence full) and one beam laser in stock.  You can illustrate 2b by going to cargo mode, putting one of the blasters into storage, then trying to buy the beam laser or a battery pack.

[Caveat Emptor~setup2.txt](https://github.com/endless-sky/endless-sky/files/4881438/Caveat.Emptor.setup2.txt)

The changes include this:
```
 bool OutfitterPanel::CanBuy() const
 {
...
        if(!playerShip)
        {
                double mass = selectedOutfit->Mass();
-               return (!mass || player.Cargo().Free() >= mass);
+               return (!mass || player.Cargo().Free() >= mass || player.Storage());
        }
```
Which seems like it explains the above behavior exactly.  "!playerShip" seems to be identical with "in cargo mode".

So it might work to fix the definition of CanBuy.  It seems like just deleting the " || player.Storage()" part works fine; even when the buy button is greyed out, you can still put things from cargo into storage just fine, and in cargo mode, if there's no room to put a stored item into cargo, then there's no reason to have the buy button no matter what's in storage.

**Comment by petervdmeer** [comment]
@waterhouse  the " || player.Storage()" part indeed looked like a bug. I removed it.

**Comment by petervdmeer** [comment]
Waterhouses remarks on 19th May got me thinking about the desired behaviour of the various states, keys and buttons and I was thinking about making the following changes:
- Remove stock from the game completely
   - For backwards compatibility; when loading, then buy stock into planetary storage

- Buy installs from supply into selected ships (same as today)
- Buy moves from supply into cargo (if no ships are selected) (same as today)
- Install installs from cargo (same as today)
- Install installs from storage (if item is not in cargo, same as in PR now)
- Install moves from storage to cargo (if no ship is selected)
- Uninstall moves an item from installed to storage
- Uninstall moves an item from cargo towards storage (if no ship with the outfit installed is selected)
- Sell completely sells an item from cargo
- Sell completely sells an item from storage (if no items in cargo)
- Sell completely sells an item from installed (if no items in cargo and no items in storage)
- Remove the ‘t’ key binding (since storage is now tied to install (‘i’), uninstall (‘u’) and sell (‘s’))

GUI-buttons:
- Buy and Install button will work the same as today
- Uninstall button will be shown when items installed or in cargo
- Sell button will only be shown when items in storage (none in cargo or installed)

What do you think?

**Comment by waterhouse** [comment]
Well, I don't necessarily have general design expertise, but I can say I like stock because I can undo any sales or purchases as long as I haven't left the planet, without reloading a snapshot.  (Except I think you can't undo selling a ship if that ship isn't normally sold there, but I think that functionality would be nice to have.)  Having selling (either from the ship or from storage) be irrevocable would take away from that.

I've been pretty happy playing with this PR as it is.  (Though earlier remarks about discoverability stand.  They no longer bother _me_ because I've discovered the hotkeys, but may be an issue for someone else.)

One thing I have been wishing for is a "take all outfits from this ship and put them into storage" button.  Early on, that is what I want to do with weaker captured ships, and with captured alien ships; later, I want to sell most outfits and keep reactors and atomic engines, but it would suffice to dump everything into storage and only occasionally sell that storage.  But that would be a minor convenience.

Incidentally, this PR is now practically needed if one wants to store the Fusion Drive from the Emerald Sword anywhere other than the Emerald Sword itself (or your cargo bay).  The only other ships with ≥230 tons of engine capacity are the "extra engines" Kestrel variant and various unique person's ships (I think the only capturable one is the Ursa Polaris).

**Comment by petervdmeer** [comment]
Okay, new proposal based on feedback from @waterhouse (here) and @tehhowch (on Discord):
- Stock stays in the game
   - Proposal #4654 adds datastructures to allow having ships "in stock", but that proposal is not relevant for the PR here. Please feel free to review the idea there though.
- Buy installs from stock/supply into selected ships (same as today)
- Buy moves from stock/supply into cargo (if no ships are selected) (same as today)
- Buy moves from stock/supply into storage (if no space to install and no space in storage).
- Buy does not install from cargo or storage into ships.
- Install installs from cargo (same as today)
- Install installs from storage (if item is not in cargo, same as in PR now)
- Install moves from storage to cargo (if no ship is selected)
- Install does not perform buying from stock or supply.
- Uninstall moves an item from installed to storage
- Uninstall moves an item from cargo towards storage (if no ship with the outfit installed is selected)
- Uninstall does not sell into stock.
- Sell sells an item from cargo into stock
- Sell sells an item from installed into stock (if no items in cargo)
- Sell sells an item from storage into stock (if ship selected with item installed and no item in cargo)
- The ‘t’ key binding will still be removed (since storage is tied to buy, install, uninstall and sell in this proposal)
   - This should keep the feature discoverable for new players.

GUI-buttons:
- Buy and Install button will work the same as today
- Uninstall button will be shown when items installed or in cargo
- Sell button will only be shown when items in storage and none in cargo or installed

A feature like "take all outfits from this ship and put them into storage (when I sell the ship)" sounds nice, but I think that is bests handled in a separate follow-up PR, because we might want to do that in a different way (by using a preference, by keeping a list of "automatic sell items" or by something even smarter that comes out of the discussion on the follow-up PR).

**Comment by waterhouse** [comment]
That sounds pretty good.  A couple of clarifications:

> Sell sells an item from storage into stock (if ship selected with item installed and no item in cargo)

I assume this should be "Sell sells an item from storage into stock (if *no* ship selected with item installed, and no item in cargo)".

> Sell button will only be shown when items in storage and none in cargo or installed

Should that be "shown when items in storage *or* in cargo or installed"?

It seems to support every workflow I can think of.  To buy into storage when you have room in your ships and cargo, you use cargo mode and press "buy" followed by "uninstall".  To uninstall items and put them into cargo, you do have to switch to cargo mode in between, which is a pain... but probably relatively rare...

Ok, let's analyze with brute force, because why not.  Locations we might want to move outfits between are: supply/stock ("stock"), storage, cargo, ship A, ship B.  That would be 5 sources x 4 destinations, but any operation with ship A is equivalent to an operation with ship B except when going cross-ship; so really we have 1 + 4 x 3 = 13 combinations.  Let's go.
- Ship A -> Ship B: select ship A, either sell or uninstall, select ship B, either buy or install (respectively)
- Stock -> Storage: If a ship or cargo has no room, select it and buy; otherwise, select cargo mode, buy, and uninstall
- Stock -> Cargo: select cargo mode, buy
- Stock -> Ship: select ship, buy
- Storage -> Stock: If cargo has none, then select it (or a ship with none) and sell; otherwise, select cargo mode, sell, and install
- Storage -> Cargo: select cargo mode, install
- Storage -> Ship: select ship, install
- Cargo -> Stock: sell (no matter what's selected)
- Cargo -> Storage: select cargo mode or a ship with none of the outfit, uninstall
- Cargo -> Ship: select ship, install
- Ship -> Stock: If cargo has none, then select ship and sell; otherwise, select ship, sell, and install
- Ship -> Storage: select ship, uninstall
- Ship -> Cargo: select ship, uninstall or sell, select cargo mode, install or buy

Storage -> Stock, Stock -> Storage, and Ship -> Stock get kind of unintuitive when you have instances of the item in cargo that you want to keep, but I'd say that's pretty rare and it's not too bad.  The worst transaction seems to be Ship -> Cargo, which now always requires two operations and a "cargo mode" switch in between, when previously it required a single keypress.  (Technically, Ship A -> Ship B requires the same amount of effort, but that seems irreducible.)

But I think Ship -> Cargo is also rare—for me, the main use of it has been taking outfits from my escorts and putting them into cargo for the Remnant "bring me 2 heavy lasers/YV shield generators/etc." missions. The fact that there was a hotkey devoted to Ship -> Cargo makes me wonder, though...  It could be worth doing a poll of how much people used the U key (and how much of that was "uninstall from ship A, reinstall to ship B").

You'll probably want to update the recently-added help text that directly says "You can also uninstall outfits directly into your cargo hold by selecting the outfit and pressing U."  (And a mission that explicitly says "Lastly, if you have an outfit installed, just tap the 'U' key [...] they will be left in your cargo hold.")  Also, on existing pilots that have already seen the help text, perhaps the new help text (if any) should consider itself not-seen-yet.

Lastly, there is the matter of bulk purchases.  Suppose the currently selected ship has room for 4 beam lasers and you do a command-buy (20x buy).  I think the command-buy should buy 4 beam lasers and stop there; only if all selected ships have no room for the outfit should a bulk purchase go into storage.  Likewise, bulk sells would only sell from storage if, before the sale starts, no selected ships or cargo have the item.  I think this is basically what's currently done, but just mentioning it.

**Comment by petervdmeer** [comment]
> > Sell sells an item from storage into stock (if ship selected with item installed and no item in cargo)
> 
> I assume this should be "Sell sells an item from storage into stock (if _no_ ship selected with item installed, and no item in cargo)".

Correct, I just fixed this in the text above.

> > Sell button will only be shown when items in storage and none in cargo or installed
> 
> Should that be "shown when items in storage _or_ in cargo or installed"?

That depends. If we want to increase visibility of the storage feature, then showing uninstall when items are installed in a ship or in cargo would work better. Maybe we should even change this to: The uninstall button is shown for installed items not locally in supply, the sell button is shown for installed items that are in local supply.

> You'll probably want to update the recently-added help text that directly says "You can also uninstall outfits directly into your cargo hold by selecting the outfit and pressing U." (And a mission that explicitly says "Lastly, if you have an outfit installed, just tap the 'U' key [...] they will be left in your cargo hold.") Also, on existing pilots that have already seen the help text, perhaps the new help text (if any) should consider itself not-seen-yet.

Agreed, this needs to be updated now.

> Lastly, there is the matter of bulk purchases. Suppose the currently selected ship has room for 4 beam lasers and you do a command-buy (20x buy). I think the command-buy should buy 4 beam lasers and stop there; only if all selected ships have no room for the outfit should a bulk purchase go into storage. Likewise, bulk sells would only sell from storage if, before the sale starts, no selected ships or cargo have the item. I think this is basically what's currently done, but just mentioning it.

Agreed, this seems the most sensible behaviour.

**Comment by Sandwichs-del** [comment]
So far I did not mind extra "storage" ships parking around the system and I even found it both easy to discover and nice that I can use the playerinfo screen to see what it stored where, but I agree it could be solved even better.

However, I see a considerable problem with the current approach, as even now I am struggling with the outfitter screen. For example, it is hard to find the right mode from installing outfits from the cargo: Intuitively that should be "Show outfits for sale == false" && "Show outfits in cargo == true", but the actual correct mode is the one that shows stuff from the shop. Adding one more location will make this even worse.

Also I rather like that you have to BUY the storage-ships, with cost proportional to storage size.
I wonder it this could be solved by offering "warehouses-ships" with a lot of outfit space,
but very heavy and with no gunports, no turrets, no weapons space and no/20 engine space.

**Comment by petervdmeer** [comment]
> Also I rather like that you have to BUY the storage-ships, with cost proportional to storage size.
> I wonder it this could be solved by offering "warehouses-ships" with a lot of outfit space,
> but very heavy and with no gunports, no turrets, no weapons space and no/20 engine space.

If we want storage to cost credits, then we could just charge the player for on-planetary storage. If we go that route, then we should also charge the player for parked ships. (But I feel both of those are for a follow-up PR if we want those features.)

**Comment by petervdmeer** [comment]
Marking as "Ready for review"; I did some brief testing with the latest updates and fixed two bugs. This PR would benefit from some more testing for the corner-cases (like trying to get outfits for free).

**Comment by waterhouse** [comment]
Mostly this is working well.  But it seems there are circumstances under which, when in cargo mode, the "buy" button and the "b" key don't work.

Below is a pilot file where, in cargo mode, if you try to buy Energy Blasters, you can't, despite ample cargo space and cash, and the Buy button being present.  I got there on be88e05ec8a8f21db708998b32f3772e2d4d2901 + the current diff (plus machine-local build config).
[Mika Dorin.txt](https://github.com/endless-sky/endless-sky/files/5432637/Mika.Dorin.txt)
Also, from cargo mode, I selected Plasma Cannons; it displayed the Install button, which makes sense because I have plenty in storage; I hit the "b" key, and it actually bought them from stock (which is unlimited on the planet) rather than moving from storage, which I actually like; I bought a bunch (still having 80 tons of cargo left), then fooled around with other items, then back at Plasma Cannons I was unable to buy any more.

Earlier, I _have_ been able to buy Energy Blasters directly from stock into cargo—I think at this very landing on the planet.  I don't know what was the key change from then.

I haven't looked into the logic to trace exactly why this stuff is happening.

**Comment by petervdmeer** [comment]
> Mostly this is working well. But it seems there are circumstances under which, when in cargo mode, the "buy" button and the "b" key don't work.

@waterhouse Thanks for checking, I had placed some brackets wrong in the check if items are in stock. Your savegame was helpful for pinpointing the issue quickly.

> Trying again, it looks like, (in cargo mode) after buying a bunch and then selling a single Plasma Cannon, the "b" key stops buying.

Yes, the problem would have appeared if there was a change in stock. Selling a single Cannon could trigger that.

> Earlier, I _have_ been able to buy Energy Blasters directly from stock into cargo—I think at this very landing on the planet. I don't know what was the key change from then.

This should now also work again.

**Comment by Sandwichs-del** [comment]
> If we want storage to cost credits, then we could just charge the player for on-planetary storage. If we go that route, then we should also charge the player for parked ships.

Sure. But I don't like the idea of _renting_ a storage. I'd like to _buy_ it.
Yes, maybe I should buy/rent storage for parked ships as well. I guess we may need both?

**Comment by Bardagh** [comment]
Any chance you could finish this? Looks like you just need to fix Mac OS compatibility and a minor merge conflict, and I could *really* use this tool <3

**Comment by petervdmeer** [comment]
> Any chance you could finish this? Looks like you just need to fix Mac OS compatibility and a minor merge conflict, and I could _really_ use this tool <3

The Mac OS test failed earlier because the test itself was unstable. Were you referring to that test, or do you see anything else that still needs to be done for this PR for Mac OS?

Merge conflict is resolved, but test_windows-unit is currently failing on master, so it was expected to also fail on this PR. The issue is actively being discussed on Discord and the Linux-version of those unit tests does pass, so I expect that issue to also go away after it is fixed in master.

**Comment by waterhouse** [comment]
I've been playing with this for a while.  I think the only surprising thing is that, when a ship is selected, "b" will install stuff if it's there, but in cargo mode, "b" will not (and even if there is stock, having an item in storage means that "b" won't do anything).  I imagine it would be good if, in the latter case, "b" did install what was there.

Other than that, it seems good.

**Review by Amazinite — CHANGES_REQUESTED** [review]
Looks great! I only noticed one bug involving being unable to move outfits from storage to cargo, having to first shuffle them through stock to get them there, but otherwise everything seems to be working as intended.

File: source/ShopPanel.cpp [hunk 1]
```diff
@@ -353,7 +353,8 @@ void ShopPanel::DrawButtons()
 	
 	Point buyCenter = Screen::BottomRight() - Point(210, 25);
 	FillShader::Fill(buyCenter, Point(60, 30), back);
-	string BUY = (playerShip && selectedOutfit && player.Cargo().Get(selectedOutfit)) ? "_Install" : "_Buy";
+	string BUY = ((playerShip && selectedOutfit && player.Cargo().Get(selectedOutfit))
+		|| (player.Storage() && player.Storage()->Get(selectedOutfit))) ? "_Install" : "_Buy";
```
[the remark this answers is not in the document]
Comment (Amazinite) [reply source/ShopPanel.cpp:357]: Actually, I just fiddled around with this some more and if I press i it WILL install from storage to cargo, so it's just a matter of the button not working.

File: source/MapPanel.cpp [hunk 2]
```diff
@@ -54,8 +55,11 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 using namespace std;
 
 namespace {
-	// Log how many player ships are in a given system, tracking if they are parked or in-flight.
-	void TallyEscorts(const vector<shared_ptr<Ship>> &escorts, map<const System *, pair<int, int>> &locations)
+	// Log how many player ships and stored outfits are in a given system, tracking for
+	// ships if they are parked or in-flight.
+	void TallyEscortsAndOutfits(const vector<shared_ptr<Ship>> &escorts,
+		const std::map<const Planet *, CargoHold> &outfits,
+		map<const System *, pair<pair<int, int>, int>> &locations)
```
[the remark this answers is not in the document]
Comment (petervdmeer) [reply source/MapPanel.cpp:62]: Good idea. I added it to the comment above the function.

**Comment by petervdmeer** [comment]
> I've been playing with this for a while. I think the only surprising thing is that, when a ship is selected, "b" will install stuff if it's there, but in cargo mode, "b" will not (and even if there is stock, having an item in storage means that "b" won't do anything). I imagine it would be good if, in the latter case, "b" did install what was there.

Thanks for testing this again. I just fixed the UI button (to ensure that 'i' is sent when the label says Install).
Cargo mode is now indeed a bit more strict than ship-installer mode; buying only works if there is something for sale and installing only works if there is something in storage.

File: source/ShopPanel.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (petervdmeer) [reply source/ShopPanel.cpp]: I just made a change to make the button give the correct key-code.

[the remark this answers is not in the document]
Comment (petervdmeer) [reply source/ShopPanel.cpp]: Thanks for the extensive testing. I went with the suggestion to label the button "Cargo". I tried "to Cargo" first, but either the text was too big or the button too small.

[the remark this answers is not in the document]
Comment (petervdmeer) [reply source/ShopPanel.cpp]: The 'c' key now also works the same as the 'i' key. (So the cargo label also has a matching key.)

**Review by Amazinite — APPROVED** [review]

**Comment by Amazinite** [comment]
Looking great! The only thing that needs changed now is the help dialog for outfitter 3. It currently states that uninstalling with the U key moves outfits to your cargo, when with this PR it will actually move them to your storage. That dialog should be changed to explain to the player how to move items to storage. 

Also, it might be worth updating the UI to add a "Show outfits in storage" checkbox to the bottom left of the outfitter that when checked would allow for purchasing directly into storage, as currently the only way to put outfits in storage is to either buy them into cargo or installed first and then move them into storage from there. Although, this can be a separate PR.
image

**Comment by petervdmeer** [comment]
I updated the help-texts and let's indeed keep the option to buy directly into storage for a separate PR.

File: data/help.txt [hunk 3]
```diff
@@ -83,10 +83,11 @@ help "outfitter"
...
+	`Outfits that are in your cargo hold can be (re)installed at any outfitter without cost.`
 
 help "outfitter 3"
-	`You can also uninstall outfits directly into your cargo hold by selecting the outfit and pressing U. This can be useful for if you want to shuffle outfits between ships or simply try on different outfits. Outfits that are in your cargo hold can be installed at any outfitter without cost.`
-	`If you sell an outfit at an outfitter, then you can immediately buy it back for the same price that you sold it, so long as you do not leave the planet. This means that you do not necessarily need to worry about uninstalling outfits into cargo when you are changing outfits around.`
+	`You can also uninstall outfits directly into planetary storage by selecting the outfit and pressing U. This can be useful for if you want to shuffle outfits between ships, store outfits for future use or simply try on different outfits. Outfits that are in planetary storage can (re)installed at any outfitter without cost. Outfits in storage can also be transferred to cargo when you go to 'purchase to cargo' mode.`
```
Comment (Amazinite) [inline data/help.txt:89]:
```suggestion
	`You can also uninstall outfits directly into planetary storage by selecting the outfit and pressing U. This can be useful for if you want to shuffle outfits between ships, store outfits for future use, or simply try on different outfits. Outfits in storage can also be transferred to cargo when you go to 'purchase to cargo' mode. Outfits that are in planetary storage or cargo can be (re)installed at any outfitter without cost.`
```

**Comment by mimikri** [comment]
hello,
the storage is awsome!
but there seems to be no option to transfer outfits from stock to storage, 
once there are items of this type in storage already 
without emptying the storage before equipping/buying the outfits from stock is not possible 
and forces a loss of the outfits once departing.

image
